src/example-types/metricGroups/pivot.js

```diff
@@ -31,6 +31,49 @@ let aggsForValues = (node, schema) =>
     }))
   )(node)
 
+let maybeWrapWithFilterAgg = ({ query, filters, aggName }) =>
+  _.isEmpty(filters)
+    ? query
+    : {
+        aggs: {
+          [aggName]: {
+            filter: { bool: { must: filters } },
+            ...query,
+          },
+        },
+      }
+
+let getSortAgg = async ({ node, sort, schema, getStats }) => {
+  if (!sort) return
+
+  let filters = _.compact(
+    await Promise.all(
+      F.mapIndexed((column, i) => {
+        let filter = lookupTypeProp(_.stubFalse, 'drilldown', column.type)
+        let drilldown = sort?.columnValues[i]
+        return drilldown && filter({ drilldown, ...column }, schema, getStats)
+      }, node.columns)
+    )
+  )
+
+  if (_.size(filters)) {
+    let valueNode = node.values[sort.valueIndex ?? 0]
```

Comment:
If `valueIndex` is null, we should either omit `metric` and sort on the filter bucket's `doc_count` (which could happen by clicking on a count column) or throw an error. Defaulting to the first value is confusing (and dangerous if there are no values).